app/connectors/MessageConnector.scala

```diff
@@ -21,13 +21,18 @@ import javax.inject.Inject
 import uk.gov.hmrc.http.{HeaderCarrier, HttpReads}
 import uk.gov.hmrc.play.bootstrap.http.HttpClient
 
-import scala.concurrent.{ExecutionContext}
-import scala.xml.NodeSeq
+import scala.concurrent.ExecutionContext
 
 class MessageConnector @Inject()(http: HttpClient, appConfig: AppConfig) {
 
-  def post(message: NodeSeq)(implicit hc: HeaderCarrier, ec: ExecutionContext) = {
-    val url = appConfig.traderAtDestinationUrl + "/transit-movements-trader-at-destination/movements/message-notification"
-    http.POSTString(url, message.toString)(HttpReads.readRaw, implicitly, implicitly)
+  def post(message: String, arrivalId: Option[String])(implicit hc: HeaderCarrier, ec: ExecutionContext) = {
+    arrivalId match {
+      case Some(id) =>
+        val url = appConfig.traderAtDestinationUrl + "/common-transit-convention-trader-at-destination/message-notification/" + id
+        http.POSTString(url, message)(HttpReads.readRaw, implicitly, implicitly)
+      case None =>
+        val url = appConfig.traderAtDestinationUrl + "/common-transit-convention-trader-at-destination/message-notification"
+        http.POSTString(url, message)(HttpReads.readRaw, implicitly, implicitly)
+    }
```

Comment:
I agree with this being moved into ArrivalConnector. I think it could be refactored into a submitMessageToDownstream method. It could take in HTTP method type, message and optional arrival id parameters.

Reply:
I'd personally consider doing something like that internally at most. There is some value in having the external methods be less generic. Plus it means we don't end up with a single monster-method. Something like SubmitAdditionalDetails makes sense. (since a lot of our stuff will be a POST of a message to an existing arrival) as well as UpdateArrival.

To me, I think it'd work best if we create a method for each endpoint downstream, that's only 7, and based on the documentation there is no real duplication there.